Add aarch64 support documentation for AlmaLinux 9 and Rocky Linux 9 on aarch64

middelkoopt opened the pull request:
* Add documentation for AlmaLinux 9 and Rocky 9 on aarch64 for Warewulf4 with Slurm
* Initial development and testing by Mike Renfro and students

middelkoopt committed:
feat(docs): add almalinux9 and rocky9 aarch64 warewulf4 slurm recipe

Add documentation for installing OpenHPC on AlmaLinux 9 and Rocky Linux 9
on aarch64 architecture using Warewulf4 provisioning and SLURM. This
includes the necessary files, Makefiles, and steps for the installation
process.

Assisted-by: Gemini 2.5 Flashopy AlmaLinux and Rocky Linux recipes from x86_64 to aarch64

Signed-off-by: Timothy Middelkoop <tmiddelkoop@internet2.edu>

middelkoopt committed:
fix(docs): update aarch64 install guides and spec file

This commit updates the aarch64 install guides to correctly
specify the architecture. Also updates the docs spec file
to include the aarch64 variants.

Assisted-by: Gemini 2.5 Flashocs: Add aarch64 support to Warewulf4

* Add documentation for AlmaLinux 9 and Rocky 9 on aarch64 for Warewulf4
  and Slurm
* Initial development and testing by Mike Renfro and students

Signed-off-by: Timothy Middelkoop <tmiddelkoop@internet2.edu>
Co-authored-by: Mike Renfro <renfro@tntech.edu>

adrianreber commented on docs/recipes/install/common/warewulf4_mkchroot.tex:
```diff
@@ -18,13 +18,14 @@ \subsubsection{Build initial BOS image} \label{sec:assemble_bos}
+[sms](*\#*) arch=ARCH
 [sms](*\#*) wwctl image exec --build=false BOSVER /bin/bash <<- EOF
-  dnf -y install http://repos.openhpc.community/OpenHPC/3/EL_9/x86_64/ohpc-release-3-1.el9.x86_64.rpm
+  dnf -y install http://repos.openhpc.community/OpenHPC/3/EL_9/${arch}/ohpc-release-3-1.el9.${arch}.rpm
```
Why not directly add `ARCH` to this line? Why first setting it as a variable and then using the variable?

middelkoopt replied:
The parsing script to generate the `recipe.sh` does not handle substitution across newlines or heredocs.  I think there is a solution for that but I forget where.  I did not like it either. but I pushed the PR out as I am not sure how much time I have to work on it in the next few days and was hoping someone might know the solution for it.

middelkoopt replied:
Well that was easier than I expected.  Pushed a new version.  Note the changes in `parse_doc.pl`

adrianreber replied:
I don't see changes to `parse_doc.pl` in this PR.

middelkoopt replied:
Just pushed for real this time.  Well that's embarrassing, forgot to add files to the index before pushing.  I just caught it as well when checking to make sure there was not any side-effects of the `parse_doc.pl` using the build artifact.

adrianreber commented:
First tests are looking good. I am running a couple more test runs and then I will merge this PR. Thanks.

Also thanks to @mikerenfro

@middelkoopt You could add the following line to the commits messages: `Co-authored-by: Mike Renfro <renfro@tntech.edu>`.

middelkoopt commented:
> @middelkoopt You could add the following line to the commits messages: `Co-authored-by: Mike Renfro <renfro@tntech.edu>`.

Done.

middelkoopt commented:
I just got my virtual arm cluster up and running and it passed my tests as well for AlmaLinux and RockyLinux.

adrianreber committed:
fix(ci): correct grep expression in parse doc test

The grep expression in the parse doc test was incorrect, causing the
test to fail. This commit corrects the grep expression to ensure that
the test passes.

Assisted-by: Gemini 2.5 Flash
Signed-off-by: Adrian Reber <areber@redhat.com>

adrianreber committed:
refactor(docs): use overlay import for hostfile modification

There seems to be a race condition in wwctl overlay edit when used in
combination with EDITOR=tee.

Assisted-by: Gemini 2.5 Flash
Signed-off-by: Adrian Reber <areber@redhat.com>

adrianreber commented:
With a rebase any my changes to handle a CI failure after the  rebase and your suggestion for a different approach to handle overlay files it now works. Thanks everyone.
